jit: irlower-call: Do proper loading in cgCheckRefs to eliminate truncation

cmuellner opened the pull request:
The vasm for CheckRefs distinguishes between 8-bit, 32-bit and 64-bit
masks. However it uses a 64-bit load for all cases in combination with
a movtqX. This patch changes this to use a load which does not require
a truncation for the cost of being little-endian specific (which is already
a requirement for the cgCheckRefs() function).

Testing the all set showed no regressions on ARM64.

Signed-off-by: Christoph Muellner <christoph.muellner@theobroma-systems.com>

cmuellner commented:
This is one of the optimisations derived from #7767.

Note, that cgCheckRefs() already uses the following code,
which is not endian neutral as well:

`hphp/runtime/vm/jit/irlower-call.cpp:469: v << testbim{(int8_t)mask64, bitsPtr[bitsOff], sf};`
`hphp/runtime/vm/jit/irlower-call.cpp:471: v << testlim{(int32_t)mask64, bitsPtr[bitsOff], sf};`


@mxw : please review
@dave-estes @jim-saxman @swalk-cavium : please test if possible
Notification also for @ptomsich

swalk-cavium commented:
Downloaded branch above.  It builds clean.  Starting MOP.
swalk@arm64:/data2/swalk/cmuellner/hhvm$ git branch -v
  master        035b577 Unclown publishOptFunctionInternal
* pr-2017-04-07 1c65d2e jit: irlower-call: Do proper loading in cgCheckRefs to eliminate truncation

swalk-cavium commented:
MOP finished.  I sent the results to @cmuellner directly via e-mail.  A quick glance
doesn't show any new issues.

cmuellner commented:
@mxw - I've introduced a loadtql and use it later on. I've implemented the loadtql emit code for x86 and ppc64 as well, however I did not test those. If x86 looks good, we might ping the ppc64 guys for testing as well.

mxw commented:
lgtm—will land once internal tests finish running.

Thanks for this series of PRs!
